File: 941. Valid Mountain Array.py
class Solution:
    def validMountainArray(self, arr: List[int]) -> bool:
        length=len(arr)
        counter=1
        if(length<3):
            counter=0
        else:
            for i in range(0,length-1):
                if(arr[i]>arr[i+1]):
                    break
                elif(arr[i]==arr[i+1]):
                    counter=0
                    break
            
            if(i==0 or i==length-1):
                counter=0
            else:
                for j in range(i,length-1):
                    if(arr[j]<arr[j+1] or arr[j]==arr[j+1]):
                        counter=0
                        break
        if(counter==1):
            return True
        else:
            return False

File: 941. Valid Mountain Array.java
class Solution {
    public boolean validMountainArray(int[] arr) {
        int counter=1;
        int length=arr.length;
        if(length<3)
            counter=0;
        else
        {
            int i,j=0;
            for(i=0;i<length-1;i++)
            {
                if(arr[i]>arr[i+1])
                    break;
                else if(arr[i]==arr[i+1])
                {
                    counter=0;
                    break;
                }
            }
            if(i==0 || i==length-1)
                counter=0;
            else
            {
                for(i=i;i<length-1;i++)
                {
                    if(arr[i]<arr[i+1] || arr[i]==arr[i+1])
                    {
                        counter=0;
                        break;
                    }
                }
            }
        }
        if(counter==1)
            return true;
        else
            return false;
    }
}
